From 714a6c792c80d226fc0c069513e48bd0b36a0c87 Mon Sep 17 00:00:00 2001
From: eko5624 <562433403@qq.com>
Date: Thu, 23 Feb 2023 14:54:37 +0800
Subject: [PATCH] workaround 'utimensat' is only available on macOS 10.13 or newer

---
 c/tools/brotli.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/c/tools/brotli.c b/c/tools/brotli.c
index 102a87a7..337f5a5e 100644
--- a/c/tools/brotli.c
+++ b/c/tools/brotli.c
@@ -75,9 +75,7 @@ static int ms_open(const char* filename, int oflag, int pmode) {
 #endif  /* defined(_WIN32) */
 
 #if defined(__APPLE__) && !defined(_POSIX_C_SOURCE)
-#define HAVE_UTIMENSAT 1
-#define ATIME_NSEC(S) ((S)->st_atimespec.tv_nsec)
-#define MTIME_NSEC(S) ((S)->st_mtimespec.tv_nsec)
+#define HAVE_UTIMENSAT 0
 #elif defined(_WIN32) || !defined(AT_SYMLINK_NOFOLLOW)
 #define HAVE_UTIMENSAT 0
 #else
