From 4843a297dbd7745bc6b6fa9efd43e19bc947e673 Mon Sep 17 00:00:00 2001
From: eko5624 <562433403@qq.com>
Date: Tue, 17 Jan 2023 07:35:43 +0800
Subject: [PATCH] Fix bundle

---
 TOOLS/dylib-unhell.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/TOOLS/dylib-unhell.py b/TOOLS/dylib-unhell.py
index b96e28d28c3..3d2e0e22227 100755
--- a/TOOLS/dylib-unhell.py
+++ b/TOOLS/dylib-unhell.py
@@ -57,7 +57,7 @@ def libraries(objfile, result = dict()):
 
     for lib in libs_list:
         if lib not in result:
-            libraries(lib, result)
+            lib = lib.replace("@loader_path", "/usr/local/lib"); libraries(lib, result)
 
     return result
 
