From 1fcf1ce6726f62357416a793fd46b38dd64f03d1 Mon Sep 17 00:00:00 2001
From: eko5624 <562433403@qq.com>
Date: Wed, 1 Feb 2023 00:24:28 +0800
Subject: [PATCH 1/2] Fix not support excluding libs

---
 lib/jxl.cmake | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/lib/jxl.cmake b/lib/jxl.cmake
index accc2b54e7..e3c6cc77ab 100644
--- a/lib/jxl.cmake
+++ b/lib/jxl.cmake
@@ -592,13 +592,6 @@ set_target_properties(jxl_dec PROPERTIES
   LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
   RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
 
-# Check whether the linker support excluding libs
-set(LINKER_EXCLUDE_LIBS_FLAG "-Wl,--exclude-libs=ALL")
-include(CheckCSourceCompiles)
-list(APPEND CMAKE_EXE_LINKER_FLAGS ${LINKER_EXCLUDE_LIBS_FLAG})
-check_c_source_compiles("int main(){return 0;}" LINKER_SUPPORT_EXCLUDE_LIBS)
-list(REMOVE_ITEM CMAKE_EXE_LINKER_FLAGS ${LINKER_EXCLUDE_LIBS_FLAG})
-
 # Add a jxl.version file as a version script to tag symbols with the
 # appropriate version number. This script is also used to limit what's exposed
 # in the shared library from the static dependencies bundled here.
@@ -617,10 +610,6 @@ foreach(target IN ITEMS jxl jxl_dec)
   # This hides the default visibility symbols from static libraries bundled into
   # the shared library. In particular this prevents exposing symbols from hwy
   # and skcms in the shared library.
-  if(LINKER_SUPPORT_EXCLUDE_LIBS)
-    set_property(TARGET ${target} APPEND_STRING PROPERTY
-        LINK_FLAGS " ${LINKER_EXCLUDE_LIBS_FLAG}")
-  endif()
 endforeach()
 
 # Only install libjxl shared library. The libjxl_dec is not installed since it

From 632c871eb285a9efee695775e3d26a4dfb3a4d64 Mon Sep 17 00:00:00 2001
From: eko5624 <562433403@qq.com>
Date: Fri, 24 Feb 2023 09:23:40 +0800
Subject: [PATCH 2/2] fix operator delete unavailable on macos 10.11

---
 lib/CMakeLists.txt | 1 -
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 1668c98455..b6ea922ab9 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -103,7 +104,6 @@ if (WIN32)
   endif()  # Clang
 else()  # WIN32
   list(APPEND JPEGXL_INTERNAL_FLAGS
-    -fsized-deallocation
     -fno-exceptions
 
     # Language flags
