From aa509cebe1ec56b9bdbe683871301a078c357a75 Mon Sep 17 00:00:00 2001
From: eko5624 <562433403@qq.com>
Date: Fri, 17 Mar 2023 09:48:28 +0800
Subject: [PATCH] Fix redefinition of '_mm512_cvtsi512_si32' on later than mac-10.14

---
 lib/jxl/enc_fast_lossless.cc | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/lib/jxl/enc_fast_lossless.cc b/lib/jxl/enc_fast_lossless.cc
index 2e8be1e177..2d7519a38e 100644
--- a/lib/jxl/enc_fast_lossless.cc
+++ b/lib/jxl/enc_fast_lossless.cc
@@ -30,16 +30,6 @@
 #elif (defined(__x86_64__) || defined(_M_X64)) && !defined(_MSC_VER)
 #include <immintrin.h>
 
-// manually add _mm512_cvtsi512_si32 definition on MacOS Mojave
-// copied from gcc 11.1.0 include/avx512fintrin.h line 14367-14373
-#if __APPLE__ && defined(__clang__) && defined(__MAC_10_0)
-inline int __attribute__((__gnu_inline__, __always_inline__, __artificial__))
-_mm512_cvtsi512_si32(__m512i __A) {
-  __v16si __B = (__v16si)__A;
-  return __B[0];
-}
-#endif
-
 // TODO(veluca): MSVC support for dynamic dispatch.
 #if defined(__clang__) || defined(__GNUC__)
