--- a/src/dispatch.c.org	2023-09-05 13:15:18
+++ b/src/dispatch.c	2023-09-05 13:17:47
@@ -405,6 +405,20 @@
 
         // Integer math has a good chance of caring about precision
         ADD(pre, "precision highp int; \n");
+    }
+
+    // textureLod() doesn't work on external/rect samplers, simply disable
+    // LOD sampling in this case. We don't currently support mipmaps anyway.
+    for (int i = 0; i < sh->descs.num; i++) {
+        if (pass_params->descriptors[i].type != PL_DESC_SAMPLED_TEX)
+            continue;
+        pl_tex tex = sh->descs.elem[i].binding.object;
+        if (tex->sampler_type != PL_SAMPLER_NORMAL) {
+            ADD(pre, "#define textureLod(t, p, b) texture(t, p) \n"
+                     "#define textureLodOffset(t, p, b, o)    \\\n"
+                     "        textureOffset(t, p, o)            \n");
+            break;
+        }
     }
 
     // Add all of the push constants as their own element
